### PR TITLE
sheffield-Armin Nouri-Structuring-week two

### PR DESCRIPTION
**Pull request by ArminNouri98**
## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

### CODE REVIEW

File: Sprint-2/errors/0.js [hunk 1]
```diff
@@ -4,6 +4,8 @@
 // interpret the error message and figure out why an error is occurring
 
 function capitalise(str) {
-  let str = `${str[0].toUpperCase()}${str.slice(1)}`;
-  return str;
+  // let str = `${str[0].toUpperCase()}${str.slice(1)}`;
+  let capitalisedStr = `${str[0].toUpperCase()}${str.slice(1)}`;
```
Comment (cjyuan) [inline Sprint-2/errors/0.js:8]: Could also just return the value directly (without first assigning it to a variable).

File: Sprint-2/extend/format-time.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  currentOutput === targetOutput,
+  `current output: ${currentOutput}, target output: ${targetOutput}`
+);
+
+const currentOutput2 = formatAs12HourClock("23:00");
+const targetOutput2 = "11:00 pm";
+console.assert(
+  currentOutput2 === targetOutput2,
+  `current output: ${currentOutput2}, target output: ${targetOutput2}`
+);
```
Comment (cjyuan) [inline Sprint-2/extend/format-time.js:24]: Why not try setting up more test cases, including edge test cases like "00:00"? What do you think
`formatAs12HourClock("00:00")` and `formatAs12HourClock("12:01")` should return?

File: Sprint-2/implement/bmi.js [hunk 3]
```diff
@@ -13,3 +13,10 @@
 // Given someone's weight in kg and height in metres
 // Then when we call this function with the weight and height
 // It should return their Body Mass Index to 1 decimal place
+function calculateBMI(weight,heght){
+let squaring=heght*heght;
+let bmiResult=70/squaring;
+return bmiResult;
+}   
+
+console.log(calculateBMI(70, 1.73)); // Outputs: 23.4
```
Comment (cjyuan) [inline Sprint-2/implement/bmi.js:22]: Does `console.log(calculateBMI(100, 2))` yield the value you expect?

**Review by cjyuan — COMMENTED** [review]
Code looks good in general. Two of your code have bugs. Have fun fixing them!

File: Sprint-2/extend/format-time.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// This is the latest solution to the problem from the prep.
+// Your task is to write tests for as many different groups of input data or edge cases as you can, and fix any bugs you find.
+
+function formatAs12HourClock(time) {
+  const hours = Number(time.slice(0, 2));
+  const minutes = time.slice(3);
+
+  if (hours === 0) return `${12}:${minutes} am`;
```
Comment (cjyuan) [inline Sprint-2/extend/format-time.js:8]: Code is correct but why not just typed `12` (instead of ${12}`? 😆

**Review by cjyuan — COMMENTED** [review]
Looks good!

File: Sprint-2/extend/format-time.js (hunk 4 above)
Comment (ArminNouri98) [reply Sprint-2/extend/format-time.js:8]: I want to to show 12:00 am and minutes same as well

Comment (cjyuan) [reply Sprint-2/extend/format-time.js:8]: I mean, why not write 
```
`${12}:${minutes} am`
```
as
```
`12:${minutes} am`
```
Same as the pm one.
